R/dependecies.R

```diff
@@ -0,0 +1,437 @@
+#' Get dependencies for all elements in lock
+get_ordered_dependencies <- function(lock) {
+   cat(crayon::blue("Retrieving dependency info..."))
+   names(lock) <- vapply(lock, `[[`, character(1), "name")
+   get_dependencies_for_list(lock, lock, list())
+}
+
+#' Recursive function to take a list and lock and extract dependencies, sorting
+#' along the way using the combine_dependencies function.
+#' @param master_list list.  Packages to go through and parse dependencies from
+#' @param lock list.  Original list of locked packages
+#' @param previously_parsed_deps.  List of packages and their dependencies that
+#'   we have already parsed out of their respective description files
+get_dependencies_for_list <- function(master_list, lock, previously_parsed_deps) {
+  ## Start off with the master list as our set of packages
+  current_dependencies <- master_list
+  for (i in seq_along(master_list)) {
+    package <- master_list[[i]]
+
+    ## Create a new list for this package if it no dependencies have been parsed
+    ## for any version of the package.
+    if(!package$name %in% names(previously_parsed_deps)) {
+      previously_parsed_deps[[package$name]] <- list()
+    }
+    if (!paste(package$version) %in% 
```

Comment:
paste comes from my using the NA value for version to mean "latest-available"
I could just use "latest-available," since that wouldn't collide with any version.  It would still, however, not be convertible to package_version...